opentelemetry-sdk/pyproject.toml

```diff
@@ -41,6 +41,14 @@ sdk = "opentelemetry.sdk.environment_variables"
 [project.entry-points.opentelemetry_id_generator]
 random = "opentelemetry.sdk.trace.id_generator:RandomIdGenerator"
 
+[project.entry-points.opentelemetry_traces_sampler]
+always_on = "opentelemetry.sdk.trace.sampling:ALWAYS_ON"
+always_off = "opentelemetry.sdk.trace.sampling:ALWAYS_OFF"
+parentbased_always_on = "opentelemetry.sdk.trace.sampling:DEFAULT_ON"
+parentbased_always_off = "opentelemetry.sdk.trace.sampling:DEFAULT_OFF"
```

Comment:
>The fix also includes some related code changes that were necessary to make this work, such as handling sampling arg correctly.

The right side of the `:` should be invokable. In the above syntax you used the `ALWAYS_ON`, `ALWAYS_OFF`, `DEFAULT_ON`, `DEFAULT_OFF` for entry point which are all instances of invokable `StaticSampler`/`ParentBased` but the following two are invokable. For this reason, you had to change the existing code to make this work. Ideally, this should not use the instances of the Sampler in entrypoint.

Reply:
Do you mean I'm providing here not invocable entities on the right side? So should the entrypoints include only invocable sampler factories? I'm not sure I understood what is the desired state here.

Reply:
Yes (for the first four), and Yes.

Reply:
Hey @srikanthccv - any chance we could get this in ready for the next release?

Many thanks!

Reply:
@matej-g I am trying to understand your new change. Does it still run into the `RuntimeError`, and the exception catch block gets executed, which means this doesn't really change anything? I was thinking we could keep your original fix and introduce a private factory for `AlwaysSample`, `NeverSample` etc.. so the expectation of a sampler factory for entry point will not be an issue. 

Something like the following in the [sampling.py](https://github.com/open-telemetry/opentelemetry-python/blob/f05fa772e65b72a1fdc0a40e142e9342dee013f3/opentelemetry-sdk/src/opentelemetry/sdk/trace/sampling.py)

```py
class _AlwaysOff(StaticSampler):
    def __init__(self):
        super().__init__(Decision.DROP)
    
class _AlwaysOn(StaticSampler):
    def __init__(self):
        super().__init__(Decision.RECORD_AND_SAMPLE)
    
class _DefaultOff(ParentBased):
    def __init__(self):
        super().__init__(ALWAYS_OFF)
    
class _DefaultOn(ParentBased):
    def __init__(self):
        super().__init__(ALWAYS_ON)
```

And symbols could be private but we can keep them private for the scope of this issue.

And the entry point could be
```
always_on = "opentelemetry.sdk.trace.sampling:_AlwaysOn"
always_off = "opentelemetry.sdk.trace.sampling:_AlwaysOff"
parentbased_always_on = "opentelemetry.sdk.trace.sampling:_DefaultOn"
parentbased_always_off = "opentelemetry.sdk.trace.sampling:_DefaultOff"
traceidratio = "opentelemetry.sdk.trace.sampling:TraceIdRatioBased"
parentbased_traceidratio = "opentelemetry.sdk.trace.sampling:ParentBasedTraceIdRatio"
```

Reply:
If that is the way to go, I can do that.

Reply:
I can't think of another better way. And without that, I think the current doesn't really fix the original issue.